### DISCUSSION TITLE
Specify User Agent string in AWS SDK for Rust

### DISCUSSION THREAD

**Discussion by trevorstr** [awslabs/aws-sdk-rust#1125] [General]
I'd like to override the User Agent string for my application in the AWS SDK. In the AWS CloudTrail logs, I need to be able to filter API calls coming from a specific application. The easiest way I can see to do this would be to specify a custom User Agent header, so I can filter on that field. How can I accomplish this with the AWS SDK for Rust?

**Comment by aajtodd** [comment]
Rather than wholesale override the User Agent header you can try giving the configuration an application name which will be added to the `User Agent` header.

e.g.

```rust
    let config = aws_config::defaults(BehaviorVersion::latest())
        .app_name(AppName::new("my-application").expect("valid app name"))
        // ... other settings
        .load()
        .await;

     let s3 = aws_sdk_s3::Client::new(&config);

```

This will add `app/my-application` to the `User-Agent` header used by the SDK.

See [ConfigLoader.app_name](https://docs.rs/aws-config/latest/aws_config/struct.ConfigLoader.html#method.app_name).

**Comment by trevorstr** [comment]
Cool, that should work, thanks!
